File: app/src/main/java/com/evaluation/navigation/Navigator.java
package com.evaluation.navigation;

import android.app.FragmentTransaction;
import android.view.ViewGroup;

import androidx.annotation.Nullable;

import com.evaluation.fragment.BaseFragment;
import com.evaluation.fragment.DetailFragment;
import com.evaluation.fragment.MainFragment;
import com.evaluation.retrofitkt.MainActivity;
import com.evaluation.retrofitkt.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * @author Vladyslav Havrylenko
 * @since 09.03.2020
 */
public class Navigator {

    private MainActivity mActivity;

    @BindView(R.id.fragment)
    ViewGroup mFragment;

    public void init(MainActivity activity) {
        mActivity = activity;
        ButterKnife.bind(this, mActivity);
    }

    public void showMainFragment() {
        MainFragment mainFragment = MainFragment.newInstance();
        showContentFragment(mainFragment);
    }

    public void showDetailFragment(int assetId) {
        DetailFragment detailFragment = DetailFragment.newInstance(assetId);
        showContentFragment(detailFragment);
    }

    private void showContentFragment(BaseFragment contentFragment) {
        FragmentTransaction fragmentTransaction = mActivity.getFragmentManager()
                .beginTransaction()
                .replace(R.id.fragment, contentFragment, contentFragment.TAG)
                .addToBackStack(contentFragment.TAG);
        fragmentTransaction.commitAllowingStateLoss();
    }

    @Nullable
    public BaseFragment getContentFragment() {
        return (BaseFragment) mActivity.getFragmentManager().findFragmentById(R.id.fragment);
    }

}

File: app/src/main/java/com/evaluation/navigation/Navigator.kt
@file:Suppress("DEPRECATION")

package com.evaluation.navigation

import com.evaluation.fragment.BaseFragment
import com.evaluation.fragment.DetailFragment
import com.evaluation.fragment.MainFragment
import com.evaluation.retrofitkt.MainActivity
import com.evaluation.retrofitkt.R
import javax.inject.Inject

/**
 * @author Vladyslav Havrylenko
 * @since 09.03.2020
 */
class Navigator @Inject constructor() {

    lateinit var mActivity: MainActivity

    fun init(activity: MainActivity) {
        mActivity = activity
    }

    fun showMainFragment() {
        showContentFragment(MainFragment.newInstance())
    }

    fun showDetailFragment(assetId: Int) {
        showContentFragment(DetailFragment.newInstance(assetId))
    }

    private fun showContentFragment(contentFragment: BaseFragment) {
        val fragmentTransaction = mActivity.supportFragmentManager
            .beginTransaction()
            .replace(R.id.fragment, contentFragment, contentFragment.TAG)
            .addToBackStack(contentFragment.TAG)
        fragmentTransaction.commitAllowingStateLoss()
    }

    val contentFragment: BaseFragment?
        get() = mActivity.fragmentManager.findFragmentById(R.id.fragment) as BaseFragment
}